CMakeLists.txt

```diff
@@ -11,9 +7,13 @@ endif()
 file(READ ${CMAKE_CURRENT_SOURCE_DIR}/scripts/version.txt ver)
 project(doctest VERSION ${ver} LANGUAGES CXX)
 
+set(CMAKE_CXX_STANDARD "11")
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
using ``CMAKE_CXX_STANDARD`` is the correct way to do this in older CMake versions.
These lines should be at the beginning (before any ``add_library/executable``) because they are inherited. So after ``project`` is the best place to put it.

Reply:
If the minimum version requirement is `3.1`, then these variables can be dropped, because [`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) and the others were introduced in `3.1`.  
Regardless, [`target_compile_features`](https://cmake.org/cmake/help/latest/command/target_compile_features.html) would be the idiomatic way to set a minimum standard requirement on a target, as that automatically propagates and doesn't assume certain directory structures.  
The less traditional CMake-isms there in the root lists file, the easier it will be to trivially pull in doctest as a dependency in modern CMake.